superset/views/core.py

```diff
@@ -319,49 +318,37 @@ def form_get(self, form):
         form.infer_datetime_format.data = True
         form.decimal.data = '.'
         form.if_exists.data = 'append'
-        all_datasources = (
-            db.session.query(
-                models.Database.sqlalchemy_uri,
-                models.Database.database_name)
-            .all()
-        )
-        form.con.choices += all_datasources
 
     def form_post(self, form):
-        def _upload_file(csv_file):
-            if csv_file and csv_file.filename:
-                filename = secure_filename(csv_file.filename)
-                csv_file.save(os.path.join(config['UPLOAD_FOLDER'], filename))
-                return filename
-
         csv_file = form.csv_file.data
-        _upload_file(csv_file)
+        form.csv_file.data.filename = secure_filename(form.csv_file.data.filename)
+        csv_filename = form.csv_file.data.filename
+        utils.upload_file(csv_file)
         table = SqlaTable(table_name=form.name.data)
-        database = (
-            db.session.query(models.Database)
-            .filter_by(sqlalchemy_uri=form.data.get('con'))
-            .one()
-        )
-        table.database = database
-        table.database_id = database.id
+        table.database = form.data.get('con')
+        table.database_id = table.database.id
         try:
-            database.db_engine_spec.create_table_from_csv(form, table)
+            table.database.db_engine_spec.create_table_from_csv(form, table)
+        except sqla_exceptions.IntegrityError:
+            os.remove(os.path.join(config['UPLOAD_FOLDER'], csv_filename))
+            flash(
+                'Table name {} already exists. Please pick another'.format(
+                    form.name.data),
+                'warning')
+            return redirect('/csvtodatabaseview/form')
+
         except Exception as e:
-            os.remove(os.path.join(config['UPLOAD_FOLDER'], csv_file.filename))
-            flash(e, 'error')
-            return redirect('/tablemodelview/list/')
+            os.remove(os.path.join(config['UPLOAD_FOLDER'], csv_filename))
+            flash(str(e), 'error')
+            return redirect('csvtodatabaseview/form')
 
-        os.remove(os.path.join(config['UPLOAD_FOLDER'], csv_file.filename))
+        os.remove(os.path.join(config['UPLOAD_FOLDER'], csv_filename))
         # Go back to welcome page / splash screen
-        db_name = (
-            db.session.query(models.Database.database_name)
-            .filter_by(sqlalchemy_uri=form.data.get('con'))
-            .one()
-        )
+        db_name = table.database.database_name
         message = _('CSV file "{0}" uploaded to table "{1}" in '
```

Comment:
while you are at it could you please make it a unicode string with u?

Reply:
Done